source/fab/application.py

```diff
@@ -58,6 +87,89 @@ def run(self, source: Path):
                 print('    found in: ' + str(filename))
                 print('    depends on: ' + str(fortran_db.depends_on(unit)))
 
+        # Start with the top level program unit
+        unit_to_process = [self._target]
```

Comment:
In order to support "configurationless build" we will need to add the ability to extract all programs from the database and use those as our targets.

Reply:
Indeed - also at some point the initial unit may be multiple units (which currently it doesn't support) and eventually it won't always be an executable